pkg/operator/controllers/azurensg/authorizer.go

```diff
@@ -0,0 +1,66 @@
+package azurensg
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/form3tech-oss/jwt-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclaim"
+	"github.com/Azure/ARO-RP/pkg/util/refreshable"
+)
+
+type credentials struct {
+	clientID     []byte
+	clientSecret []byte
+	tenantID     []byte
+}
+
+func newAuthorizer(token *adal.ServicePrincipalToken) (refreshable.Authorizer, error) {
+	p := &jwt.Parser{}
+	c := &azureclaim.AzureClaim{}
+	_, _, err := p.ParseUnverified(token.OAuthToken(), c)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, role := range c.Roles {
+		if role == "Application.ReadWrite.OwnedBy" {
+			return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalCredentials, "properties.servicePrincipalProfile", "The provided service principal must not have the Application.ReadWrite.OwnedBy permission.")
+		}
+	}
+
+	return refreshable.NewAuthorizer(token), nil
```

Comment:
I know that we talked about placing this maybe in `pkg/util/` because it's being used also in `pkg/api/validate/dynamic/serviceprincipal.go` for the ValidateServicePrincial` function to get away from duplication.  @bennerv did this refactor in his PR with SP conditions on the operator.

Reply:
I moved the auth part it to /pkg/util and I'll reach out to Ben. Also changed the code to reuse the underlying functions used in the validator, but I didn't want to reuse the validation functions. They accept the whole oc document, and I don't want to pass it all inside the cluster.